### PR TITLE
[full-ci] Use resharing capability instead of hardcoding isOcis

### PR DESCRIPTION
**Pull request by dschmidt**
## Description
This PR replaces the not-so-granular `isOcis` getter (which relies on `edition === 'reva'`) by composables which in turn read capabilities from the store. 

## Related Issue
- Unblocks https://github.com/owncloud/ocis/pull/2918

## Motivation and Context
The `edition === 'reva'` check blocks https://github.com/owncloud/ocis/pull/2918

## How Has This Been Tested?

- Tests for the new composable(s) are added
- Existing tests have been fixed

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Technical debt
- [ ] Tests

## Checklist:

- [ ] Code changes
- [x] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation ticket raised: <link> 

## Open tasks:
- [ ] Find a smarter check than the resharing capability for deciding whether the account editing should be enabled or not

### CODE REVIEW

**Comment by ownclouders** [comment]
Results for <strong>oCISTrashbinUploadMoveJourney</strong> https://drone.owncloud.com/owncloud/web/23848/69/1
The following scenarios passed on retry:
- webUITrashbinRestore/trashbinRestore.feature:279

**Comment by ownclouders** [comment]
Results for <strong>oCISSharingPublic1</strong> https://drone.owncloud.com/owncloud/web/23848/67/1
The following scenarios passed on retry:
- webUISharingPublicBasic/publicLinkCreate.feature:190

File: packages/web-app-files/src/store/actions.js [hunk 1]
```diff
@@ -14,6 +14,11 @@ import { loadPreview } from '../helpers/resource'
 import { avatarUrl } from '../helpers/user'
 import { has } from 'lodash-es'
 import { ShareTypes, SpacePeopleShareRoles } from '../helpers/share'
+import get from 'lodash-es/get'
+
+const allowSharePermissions = (getters, state) => {
```
Comment (kulmann) [inline packages/web-app-files/src/store/actions.js:19]:
```suggestion
const allowSharePermissions = (getters) => {
```

File: packages/web-runtime/src/composables/capability/useCapability.ts [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+  store: Store<any>,
+  name: string,
+  defaultValue?: T
+): ComputedRef<T> => {
+  return computed((): T => {
+    if (defaultValue) {
+      return get(store, `getters.capabilities.${name}`, defaultValue) as T
+    }
+
+    const value = get(store, `getters.capabilities.${name}`) as T
```
Comment (kulmann) [inline packages/web-runtime/src/composables/capability/useCapability.ts:16]:
```suggestion
    const value = get(store, `getters.capabilities.${name}`, defaultValue) as T
```

File: packages/web-runtime/src/composables/capability/useCapability.ts [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+import { computed, ComputedRef } from '@vue/composition-api'
+import { useStore } from 'web-pkg/src/composables'
+
+export const useCapability = <T>(
+  store: Store<any>,
+  name: string,
+  defaultValue?: T
+): ComputedRef<T> => {
+  return computed((): T => {
+    if (defaultValue) {
```
Comment (kulmann) [inline packages/web-runtime/src/composables/capability/useCapability.ts:12]: not necessary if you apply the other suggestion in this file

File: packages/web-runtime/src/composables/capability/useCapability.ts [hunk 4]
```diff
@@ -0,0 +1,34 @@
+import { Store } from 'vuex'
```
Comment (kulmann) [inline packages/web-runtime/src/composables/capability/useCapability.ts:1]: Please move to `web-pkg`.

File: packages/web-runtime/src/pages/account.vue [hunk 5]
```diff
@@ -73,21 +73,30 @@
...
 import { mapGetters } from 'vuex'
+import { useCapabilityFilesSharingResharing } from '../composables'
 export default {
   name: 'Personal',
+  setup() {
+    return {
+      // FIXME: We need a clean way to know whether account editing is enabled or not
+      // for now we abuse the files_sharing.resharing capability, as *currently*
+      // oCIS does not support that exactly like it does not support account editing
+      isAccountEditingEnabled: useCapabilityFilesSharingResharing()
```
Comment (kulmann) [inline packages/web-runtime/src/pages/account.vue:84]: Please check instead if the `apps` store module from the runtime has the `settings` app. That's a more reliable indicator for now whether we run ocis or not.

**Review by kulmann — CHANGES_REQUESTED** [review]

**Comment by kulmann** [comment]
can you please rebase? there are conflicts now. and https://github.com/owncloud/web/pull/6577 has been merged with additional `isOcis` mocks in tests

File: packages/web-runtime/src/composables/capability/useCapability.ts (hunk 2 above)
Comment (dschmidt) [reply packages/web-runtime/src/composables/capability/useCapability.ts:16]: Ah, yes, you are totally right.

File: packages/web-runtime/src/composables/capability/useCapability.ts (hunk 3 above)
Comment (dschmidt) [reply packages/web-runtime/src/composables/capability/useCapability.ts:12]: Ah, yes, you are totally right.

**Comment by ownclouders** [comment]
Results for <strong>oCISBasic</strong> https://drone.owncloud.com/owncloud/web/23885/51/1
<details><summary>:boom: The acceptance tests failed on retry. Please find the screenshots inside ...</summary>

<p>


### webUIAccount-accountInformation_feature-L10.png
  


</p></details>

**Review by kulmann — APPROVED** [review]
lgtm 👍🏻

**Pull request merged**